lib/presentation/localization/locale_inherited_widget.dart

```diff
@@ -0,0 +1,15 @@
+
+import 'package:flutter/widgets.dart';
+
+class LocaleInheritedWidget extends InheritedWidget {
+  final Function(String) onLanguageChanged;
```

Comment:
This approach is not "Flutterized" too, i am even sure that it does not work at all :)
Here's how it should work.
When you call context.dependOnInheritedWidgetOfExactType(), the Widget's Element class is marked dependent on that InheritedWidget and whenever InheritedWidget's is changed (updateShouldNotify returns true), dependant widgets are called to rebuild. InheritedWidgets should always be immutable, that means that inherited widgets do not update from within, they are created anew instead. In your widgets tree only Stateful widgets should be updated within and there should be no exceptions since Flutter is not aware of any updates within non-stateful widgets.
So in your case the language value can be stored and updated within in your _LocalizedAppState. You can check the implementation of other StatefulWidgets accessed with context, for example Navgator, to change the language in StatefulWidget from the tree decendents. Then you need to provide language to LocaleInheritedWidget and it will be spread among all the dependant widgets. That's how InheritedWidgets should be used.

Reply:
> So in your case the language value can be stored and updated within in your _LocalizedAppState.
Then you need to provide language to LocaleInheritedWidget and it will be spread among all the dependant widgets.

If I get you right - I did the same, I stored `language` in `_LocalizedAppState` and when `language` is changed `LocaleInheritedWidget` is created anew and notify its subscribers, in my case it is `MaterialAppWidget` which consumes 'language' and gets rebuild with its descendants.

Reason of appearing `final Function(String) onLanguageChanged` in `LocaleInheritedWidget` is that I need to lift state up from descendant `SettingsPage` which have UI for change the language to the parent `MaterialAppWidget` which consumes the language. As InheritedWidget is immutable I cannot reassign its field but I can use its field to change my data inside it as said Philip [here](https://youtu.be/Zbm3hjPjQMk?t=200).

Also I found [here](https://youtu.be/1t-8rBCGBYw?t=48) that for lifting state up is better to use other approaches then inherited widget, but I want to try InheritedWidget as a base for any other state management solutions.

Maybe I do not understand something here, could show or explain in more details how it will be more beautiful?

Reply:
As far as I remember I found this trick [here](https://stackoverflow.com/a/49492495/5434904) and adapt it for my needs.

Reply:
> i am even sure that it does not work at all :)

It works, I have double checked it =) I had a problem with building Flutter dependencies after updating Flutter plugin, but it is fixed already by bumping dependencies versions up.